skip the proxy if the browser supports CORS

jgravois opened the pull request.

jgravois commented:
@kneemer if you're up for reviewing this PR (and perhaps the other open one too) then I'll tag another release and submit a PR to fix the problem in the developers site too.

ngoldman commented on geotrigger.js:
```diff
@@ -222,7 +222,7 @@
-    if (this.proxy) {
+    if (this.proxy && !CORS) {
       url = this.proxy + url;
     }
```
@jgravois is there a scenario where CORS is supported but you'd still want a proxy to be used?

JerrySievert replied:
wouldn't you want to use a proxy if you're going to have the possibility of exposing credentials? (one of the cases for proxy).

jgravois replied:
i definitely understand the workflow of embedding credentials in a proxy so that end users don't have to authenticate, but in this repository the sample proxy thats included doesn't support it and the only mention of proxies is to allow for use in IE8/9.

all that being said, i could also:
* go back to injecting the logic which tests for/utilizes CORS in the editor and faker applications instead
* include a configuration option here so that developers can decide whether or not they want to force use of proxy
* do something else entirely (you'll have to explain what :smile:)

jgravois commented:
@kneemer and i agreed that it'd be best to introduce a new flag for developers in case they are interested in forcing use of the proxy. `ignoreCORS` is set to `false` by default.

i also dropped the minified version of the library from the repo and added a tiny bit more sugar to the `package.json`.  after this gets merged, and after Travis is hooked up, i'll make sure to introduce an additional step in the release process to make sure that the minified copy is included there only.

ngoldman commented:
:cool:

kneemer commented on geotrigger.js:
```diff
@@ -216,13 +217,14 @@
 
-    if (this.proxy) {
+    // use a proxy if CORS support isn't present, or developer wants to force it (to do something like pass through credentials)
+    if (this.proxy && this.ignoreCORS || !CORS) {
```
should this be (this.proxy && (this.ignoreCORS || !CORS))?

jgravois replied:
my first guess had the effect i was going after in both browsers that support CORS and ones that don't:

1. see if a proxy has even be specified, if not, skip second conditional check entirely and just don't use a proxy.
2. is ignoreCORS truthy?  if so, go ahead and use the proxy
3. if ignoreCORS is falsey, evaluate the last statement.  if no CORS support is present, use the proxy.

i think yours is:

1. see if a proxy has even be specified, if not, skip second conditional check entirely and just don't use a proxy.
2. check to see if ignoreCORS is truthy or the browser lacks CORS support simultaneously, if either, use the proxy

all that being said, i'm happy to make the suggested change because your fu is typically stronger than mine.

jgravois replied:
nm.  im totally [wrong](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence).  `&&`s are evaluated before `||`s
